sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -132,7 +134,9 @@ export class RequestResponseLink implements ReqResLink {
 
       const onAbort = (): void => {
         // safe to clear the timeout if it hasn't already occurred.
-        clearTimeout(timer);
+        if (timer != null) {
```

Comment:
the result of `setTimeout` is never falsy, right? So wouldn't this be enough?

```suggestion
        if (timer) {
```

Reply:
Well, I actually tried finding if setTimeout could ever return `0` in browsers. From what I could tell, at least most browsers start with `1` when calling setTimeout, but that it wasn't part of the spec so in theory a browser could return `0`. But I'd love to be wrong on this!

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#return_value

You're right that it doesn't appear directly in a spec, but AFAIK every browser I've ever tested managed to not mess this up, probably because enough people use simple truthy checks for this. :)